src/components/ChallengeEditor/NDAField/index.js

```diff
@@ -1,26 +1,58 @@
 import _ from 'lodash'
 import React from 'react'
 import PropTypes from 'prop-types'
+import cn from 'classnames'
 import styles from './NDAField.module.scss'
 import { DEFAULT_NDA_UUID } from '../../../config/constants'
 
 const NDAField = ({ challenge, toggleNdaRequire, readOnly }) => {
   const isRequiredNda = challenge.terms && _.some(challenge.terms, { id: DEFAULT_NDA_UUID })
+
+  if (readOnly) {
+    return (
+      <div className={styles.row}>
+        <div className={cn(styles.field, styles.col1, styles.fieldTitle)}>NDA Required :</div>
+        <div className={cn(styles.field, styles.col2)}>
+          {
+            isRequiredNda
+              ? <div>Yes</div>
+              : <div>No</div>
+          }
+        </div>
+      </div>
+    )
+  }
+
   return (
     <div className={styles.row}>
-      <div className={styles.tcCheckbox}>
-        <input
-          name='isRequiredNda'
-          type='checkbox'
-          id='isRequiredNda'
-          checked={isRequiredNda}
-          onChange={toggleNdaRequire}
-          readOnly={readOnly}
-        />
-        <label htmlFor='isRequiredNda' className={readOnly ? styles.readOnly : ''}>
-          <div>NDA Required</div>
-          <input type='hidden' />
-        </label>
+      <div className={cn(styles.field, styles.col1, styles.fieldTitle)}>NDA Required :</div>
+      <div className={cn(styles.field, styles.col2)}>
+        <div className={styles.tcRadioButton}>
+          <input
+            name='nda'
+            type='radio'
+            id='nda-yes'
+            checked={isRequiredNda}
+            onChange={toggleNdaRequire}
+          />
+          <label className={styles['tc-RadioButton-label']} htmlFor='nda-yes'>
+            <div>yes</div>
+            <input type='hidden' />
```

Comment:
@CDharmateja do we really need this "hidden" input?

Reply:
I think its not necessary. The input radio structure is like that here https://github.com/topcoder-platform/work-manager/blob/6a0579aa2b10f3345c65ec573e785db75ba6ea50/src/components/ChallengeEditor/ReviewType-Field/index.js#L31. So I have used it.